### PR TITLE
#117 - Added Error interface and BlobUnknownError

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #117 
Added `Error` interface and `BlobUnknownError` implementation.

### CODE REVIEW

File: src/main/java/com/artipie/docker/error/DockerError.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+     * Get code.
+     *
+     * @return Code identifier string.
+     */
+    String code();
+
+    /**
+     * Get message.
+     *
+     * @return Message describing conditions;
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/docker/error/DockerError.java:48]: @olegmoz please replace ";" with "." at the end of the sentence

File: src/main/java/com/artipie/docker/error/BlobUnknownError.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    @Override
+    public String message() {
+        return "blob unknown to registry";
+    }
+
+    @Override
+    public Optional<String> detail() {
+        return Optional.of(this.digest.string());
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/docker/error/BlobUnknownError.java:64]: @olegmoz hm, should not the details be a bit more informative and written in proper English?

Comment (olegmoz) [reply src/main/java/com/artipie/docker/error/BlobUnknownError.java:64]: @olenagerasimova not sure about it, I took real registry as example, see: https://mcr.microsoft.com/v2/dotnet/core/runtime/blobs/sha256:b81717fef0141577dd1588f2838d9a797e026ca20d95d0a89559a6b6af734c7b

File: src/main/java/com/artipie/docker/error/DockerError.java (hunk 1 above)
Comment (olegmoz) [reply src/main/java/com/artipie/docker/error/DockerError.java]: @olenagerasimova fixed, thanks

File: src/main/java/com/artipie/docker/error/BlobUnknownError.java (hunk 2 above)
Comment (olenagerasimova) [reply src/main/java/com/artipie/docker/error/BlobUnknownError.java:64]: @olegmoz ok, I see, thanks

**Pull request merged**